schema/object_test.go

```diff
@@ -94,6 +94,43 @@ func TestObjectUnserialization(t *testing.T) {
 	})
 }
 
+func TestObjectUnserialization_MissingFields(t *testing.T) {
+	dataMissing1 := map[string]any{
+		"field3": "Hello world!",
+	}
+	_, err := testStructSchema.Unserialize(dataMissing1)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "field is required")
+
+	dataMissing3 := map[string]any{
+		"Field1": 42,
+	}
+	_, err = testStructSchema.Unserialize(dataMissing3)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "field is required")
+}
+
+func TestObjectUnserialization_IncorrectType(t *testing.T) {
+	dataMissing1 := map[string]any{
+		"Field1": "this cannot be represented as an integer",
+		"field3": "Hello world!",
+	}
+	_, err := testStructSchema.Unserialize(dataMissing1)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "parsing")
```

Comment:
Again, can we verify that it's complaining about attempting to parse "this cannot be represented as an integer" "as an integer" rather than just that there's some "parsing" error? In context, there's not a lot of doubt; but in addition to narrowing down the detected error, I like these sort of tests to verify that the error message is formatted in a way that's both correct and useful instead of ignoring the "meat" in favor of a few keywords.

Reply:
I pushed changes that test that the data itself is present.